### PR TITLE
Unexpected behavior when managing multiple sources with empty ListProperty #154

### PR DESCRIPTION
**Pull request by Rizen59**
Unit test added

### CODE REVIEW

File: core/src/main/java/com/dooapp/fxform/controller/PropertyEditorController.java [hunk 1]
```diff
@@ -70,8 +71,10 @@ public void changed(ObservableValue observableValue, Object o, Object o1) {
                     }
                     Object newValue = propertyElementValidator.adapt(o1, adapter);
                     // check whether the value represented in the view is different from the current model value
+                    // particular case is applied for list because two different instances of empty list are considered equals
```
Comment (amischler) [inline core/src/main/java/com/dooapp/fxform/controller/PropertyEditorController.java]: Could you extract the following duplicated code (see line 118) to an external method (e.g. isSame(oldValue, newValue)) and add unit tests covering the various cases ?

File: core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java [hunk 2]
```diff
@@ -31,6 +39,33 @@
 	public AbstractFieldElement(Field field) throws FormException {
 		super();
 		this.field = field;
+		init();
+	}
+
+	protected void init() {
```
Comment (amischler) [inline core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java]: Why did you move this method from AbstractSourceElement to this class ? I think that the init() implementation should be still in AbstractSourceElement.

File: core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java [hunk 3]
```diff
@@ -50,4 +85,48 @@ public String getName() {
...
+				}
+
+				@Override
+				public void dispose() {
+					super.dispose();
+					unbind(sourceProperty());
+				}
+			};
+		}
+		return new ObjectBinding<ObservableValue<WrappedType>>() {
```
Comment (amischler) [inline core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java]: Could you add an **else** statement for better readability here ?

File: core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java (hunk 2 above)
Comment (Rizen59) [reply core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java]: We must ensure the field is set before the creation of the wrappedProperty, needed to build the right type of binding.
By adding the init implementation in the parent constructor the field will still be null.

So another solution would be to manually call the init method after the full object is built but it will many changes in the project as well as break cause breaking change to the current users. This is a thing we don't actually want.

File: core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java (hunk 3 above)
Comment (Rizen59) [reply core/src/main/java/com/dooapp/fxform/model/impl/AbstractFieldElement.java]: I can, but I'm from the side of people that think it's a better readability without the else when not needed in case of return statement :
- https://www.refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
- https://stackoverflow.com/questions/3261849/why-is-else-rarely-used-after-if-x-then-return

File: core/src/main/java/com/dooapp/fxform/controller/PropertyEditorController.java (hunk 1 above)
Comment (Rizen59) [reply core/src/main/java/com/dooapp/fxform/controller/PropertyEditorController.java]: Done!
